app/app.go

```diff
@@ -12,24 +16,38 @@ import (
 	"github.com/bnb-chain/greenfield-challenger/verifier"
 	"github.com/bnb-chain/greenfield-challenger/vote"
 	ethcommon "github.com/ethereum/go-ethereum/common"
-	"gorm.io/driver/mysql"
 	"gorm.io/gorm"
 )
 
 type App struct {
-	executor      *executor.Executor
-	eventMonitor  *monitor.Monitor
-	hashVerifier  *verifier.Verifier
-	voteProcessor *vote.VoteProcessor
-	txSubmitter   *submitter.TxSubmitter
+	executor        *executor.Executor
+	eventMonitor    *monitor.Monitor
+	hashVerifier    *verifier.Verifier
+	voteCollector   *vote.VoteCollector
+	voteBroadcaster *vote.VoteBroadcaster
+	voteCollator    *vote.VoteCollator
+	txSubmitter     *submitter.TxSubmitter
 }
 
 func NewApp(cfg *config.Config) *App {
 	db, err := gorm.Open(mysql.Open(cfg.DBConfig.DBPath), &gorm.Config{})
 	// db = db.Debug() only for debug purpose
 	if err != nil {
-		panic(fmt.Sprintf("open db error, err=%s", err.Error()))
+		panic(fmt.Sprintf("open db error, err=%+v", err.Error()))
+	}
+
+	dbConfig, err := db.DB()
+	if err != nil {
+		panic(err)
 	}
+	dbConfig.SetMaxIdleConns(20)
+	dbConfig.SetMaxOpenConns(40)
```

Comment:
Can we add these parameters into config?

Reply:
fixed